From 1c12c24364d1058ffdbb28fca72a51de85082778 Mon Sep 17 00:00:00 2001
From: Rich Felker <dalias@aerifal.cx>
Date: Fri, 5 Dec 2014 21:15:41 -0500
Subject: [PATCH] don't fail posix_spawn on failed close

the resolution of austin group issue #370 removes the requirement that
posix_spawn fail when the close file action is performed on an
already-closed fd. since there are no other meaningful errors for
close, just ignoring the return value completely is the simplest fix.
---
 src/process/posix_spawn.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/process/posix_spawn.c b/src/process/posix_spawn.c
index ae262f7..af12731 100644
--- a/src/process/posix_spawn.c
+++ b/src/process/posix_spawn.c
@@ -102,8 +102,7 @@ static int child(void *args_vp)
 			}
 			switch(op->cmd) {
 			case FDOP_CLOSE:
-				if ((ret=__syscall(SYS_close, op->fd)))
-					goto fail;
+				__syscall(SYS_close, op->fd);
 				break;
 			case FDOP_DUP2:
 				if ((ret=__sys_dup2(op->srcfd, op->fd))<0)
-- 
2.2.0

